third_party/amd/lib/TritonAMDGPUToLLVM/SchedInstructions.cpp

```diff
@@ -253,8 +230,8 @@ struct InstructionSchedHintsRewriter
 
     if (!(schedHint.getIsBufferLoadsAEnabled() &&
           schedHint.getIsBufferLoadsBEnabled())) {
-      LDBG("skipping `local-prefetch` scheduling given it needs `buffer_load` "
-           "instructions");
+      LDBG("skipping `local_prefetch` scheduling given it needs `buffer_load` "
+           "instructions.");
```

Comment:
I always want to ask, why the `local_prefetch` scheduling needs a specific type of instruction? In this particular case, why global_load does not work?

Reply:
`global_loads` involve address calculations (VALU instructions) and it has loop-carried dependencies. It turns out this messes up scheduling imposed by LLVM intrinsic calls - i.e., the compiler backend gets confused. In short, the CK-like scheduling works only with `buffer_load` instructions. Based on my experience, it is not worth it to apply `local_prefetch` to a GEMM variant which involve `global_loads`

Reply:
so buffer_load does address update with SALU instructions?
>  the compiler backend gets confused

How confused? Does it place those VALU's in the wrong place so that their latencies are exposed?